MonoGame.Framework.Content.Pipeline/TextureImporter.cs

```diff
@@ -63,168 +63,46 @@ public TextureImporter()
-                                FreeImage.UnloadEx(ref r);
-                                FreeImage.UnloadEx(ref b);
-                            }
-                            break;
-                    }
-                    break;
-
-                case FREE_IMAGE_TYPE.FIT_RGBF:
-                    {
-                        // Swap R and B channels to make it BGR, then add an alpha channel
-                        var r = FreeImage.GetChannel(fBitmap, FREE_IMAGE_COLOR_CHANNEL.FICC_RED);
-                        var b = FreeImage.GetChannel(fBitmap, FREE_IMAGE_COLOR_CHANNEL.FICC_BLUE);
-                        FreeImage.SetChannel(fBitmap, b, FREE_IMAGE_COLOR_CHANNEL.FICC_RED);
-                        FreeImage.SetChannel(fBitmap, r, FREE_IMAGE_COLOR_CHANNEL.FICC_BLUE);
-                        FreeImage.UnloadEx(ref r);
-                        FreeImage.UnloadEx(ref b);
-                        var bgra = FreeImage.ConvertToType(fBitmap, FREE_IMAGE_TYPE.FIT_RGBAF, true);
-                        FreeImage.UnloadEx(ref fBitmap);
-                        fBitmap = bgra;
-                    }
-                    break;
-
-                case FREE_IMAGE_TYPE.FIT_RGBAF:
-                    {
-                        // Swap R and B channels to make it BGRA
-                        var r = FreeImage.GetChannel(fBitmap, FREE_IMAGE_COLOR_CHANNEL.FICC_RED);
-                        var b = FreeImage.GetChannel(fBitmap, FREE_IMAGE_COLOR_CHANNEL.FICC_BLUE);
-                        FreeImage.SetChannel(fBitmap, b, FREE_IMAGE_COLOR_CHANNEL.FICC_RED);
-                        FreeImage.SetChannel(fBitmap, r, FREE_IMAGE_COLOR_CHANNEL.FICC_BLUE);
-                        FreeImage.UnloadEx(ref r);
-                        FreeImage.UnloadEx(ref b);
-                    }
-                    break;
-            }
+            fBitmap = ConvertAndSwapChannels(fBitmap, imageType);
 
             // The bits per pixel and image type may have changed
-            bpp = FreeImage.GetBPP(fBitmap);
+            uint bpp = FreeImage.GetBPP(fBitmap);
             imageType = FreeImage.GetImageType(fBitmap);
-            var pitch = (int)FreeImage.GetPitch(fBitmap);
+            var pitch = (int) FreeImage.GetPitch(fBitmap);
             var redMask = FreeImage.GetRedMask(fBitmap);
             var greenMask = FreeImage.GetGreenMask(fBitmap);
             var blueMask = FreeImage.GetBlueMask(fBitmap);
 
-            // Get the bytes from the FreeImage bitmap
-            var bytes = new byte[width * height * (bpp / 8)];
-            FreeImage.ConvertToRawBits(bytes, fBitmap, pitch, bpp, redMask, greenMask, blueMask, true);
+            // Create the byte array for the data,
+            // In case of modifying the code to generate pixel data with less than 8 bpp do not remove the casts or the size or the size will be wrong
+            var bytes = new byte[(int) (width * height * ((float) bpp / 8))];
```

Comment:
Rather than do the operation as float and convert back to int (and run the risk of floating point inaccuracy), keep it as integer all the way through, and it's easier to read.
`var bytes = new byte[(width * height * bpp) / 8];`